articles/azure-monitor/app/tutorial-custom-metrics.md

```diff
@@ -0,0 +1,370 @@
+
+In summary `GetMetric()` is the recommended approach since it does pre-aggregation, it accumulates values from all the Track() calls and sends a summary/aggregate once every minute. `GetMetric()` can significantly reduce the cost and performance overhead by sending fewer data points, while still collecting all relevant information.
+
+## Getting a TelemetryClient instance
+
+Get an instance of `TelemetryClient` from the dependency injection container in **HomeController.cs**:
+
+```csharp
+//... additional code removed for brevity
+using Microsoft.ApplicationInsights;
+
+namespace AzureCafe.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly ILogger<HomeController> _logger;
+        private AzureCafeContext _cafeContext;
+        private BlobContainerClient _blobContainerClient;
+        private TextAnalyticsClient _textAnalyticsClient;
+        private TelemetryClient _telemetryClient;
+
+        public HomeController(ILogger<HomeController> logger, AzureCafeContext context, BlobContainerClient blobContainerClient, TextAnalyticsClient textAnalyticsClient, TelemetryClient telemetryClient)
+        {
+            _logger = logger;
+            _cafeContext = context;
+            _blobContainerClient = blobContainerClient;
+            _textAnalyticsClient = textAnalyticsClient;
+            _telemetryClient = telemetryClient;
+        }
+
+        //... additional code removed for brevity
+    }
+}
+```
+
+`TelemetryClient` is thread safe.
+
+## TrackMetric
+
+> [!NOTE]
+> `Microsoft.ApplicationInsights.TelemetryClient.TrackMetric` isn't the preferred method for sending metrics. Metrics should always be pre-aggregated across a time period before being sent. Use one of the `GetMetric(..)` overloads to get a metric object for accessing SDK pre-aggregation capabilities.
+>
+> If you're implementing your own pre-aggregation logic, you can use the `TrackMetric()` method to send the resulting aggregates. If your application requires sending a separate telemetry item on every occasion without aggregation across time, you likely have a use case for event telemetry. See `TelemetryClient.TrackEvent
+(Microsoft.ApplicationInsights.DataContracts.EventTelemetry)`.
+
+Application Insights can chart metrics that aren't attached to particular events. For example, you could monitor a queue length at regular intervals. With metrics, the individual measurements are of less interest than the variations and trends, and so statistical charts are useful.
+
+To send metrics to Application Insights, you can use the `TrackMetric(..)` API. There are two ways to send a metric:
+
```

Comment:
I'd suggest to focus only on the aggregating API. The beginning of the doc already made the recommendation that it is the recommended one.

Reply:
Fixed.